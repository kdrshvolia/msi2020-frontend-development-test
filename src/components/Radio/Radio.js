import React from 'react';import PropTypes from 'prop-types';const Radio = ({ name, value, option, checked, handleChange }) => {  const onChange = () => {    handleChange(option);  };  return (    <>      <input type="radio" name={name} checked={checked} onChange={onChange} />      {value}    </>  );};Radio.propTypes = {  name: PropTypes.string.isRequired,  value: PropTypes.string.isRequired,  option: PropTypes.string.isRequired,  handleChange: PropTypes.func.isRequired,};export default Radio;