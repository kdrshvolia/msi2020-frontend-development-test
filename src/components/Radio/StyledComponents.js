import styled, { css } from 'styled-components';

const RadioWrapper = styled.div`
  padding: 0 10px;
`;

const CustomRadio = styled.input`
  &:checked,
  &:not(:checked) {
    position: absolute;
    left: -9999px;
  }
  &:checked + label,
  &:not(:checked) + label {
    position: relative;
    padding-left: 28px;
    cursor: pointer;
    line-height: 20px;
    display: inline-block;
  }
  &:not(:checked) + label:before {
    content: '';
    position: absolute;
    left: 0;
    top: 0;
    width: 20px;
    height: 20px;
    border: 2px solid #ababab;
    box-sizing: border-box;
    border-radius: 100%;
    background: #fff;
  }

  &:checked + label:before {
    border: 2px solid #333333;
    content: '';
    position: absolute;
    left: 0;
    top: 0;
    width: 20px;
    height: 20px;
    box-sizing: border-box;
    border-radius: 100%;
    background: #fff;
  }
  &:checked + label:after,
  &:not(:checked) + label:after {
    content: '';
    width: 10px;
    height: 10px;
    background: #333333;
    position: absolute;
    top: 5px;
    left: 5px;
    border-radius: 100%;
    -webkit-transition: all 0.2s ease;
    transition: all 0.2s ease;
  }

  &:not(:checked) + label:after {
    opacity: 0;
    -webkit-transform: scale(0);
    transform: scale(0);
  }
  &:checked + label:after {
    opacity: 1;
    -webkit-transform: scale(1);
    transform: scale(1);
  }
`;

export { CustomRadio, RadioWrapper };
