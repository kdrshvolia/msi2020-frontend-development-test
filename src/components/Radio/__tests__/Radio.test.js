import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Radio from '../Radio';

const option = 'search';
const isChanged = jest.fn();

describe('Radio', () => {
  it('Radio changes', () => {
    const wrapper = mount(<Radio handleChange={isChanged} option={option} />);
    const radio = wrapper.find('input[type="radio"]');
    radio.simulate('change');
    expect(isChanged).toBeCalledWith(option);
  });
  it('Renders correctly', () => {
    const rendered = renderer.create(<Radio handleChange={isChanged} option={option} />);
    expect(rendered).toMatchSnapshot();
  });
});
