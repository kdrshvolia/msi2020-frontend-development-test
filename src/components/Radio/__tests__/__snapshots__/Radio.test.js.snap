// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Renders correctly 1`] = `
<div
  className="sc-AxjAm dWVCDA"
>
  <input
    className="sc-AxirZ hxQKcv"
    onChange={[Function]}
    type="radio"
  />
   
  <label />
</div>
`;
